Cargo.toml

```diff
@@ -17,8 +17,11 @@ linux-native = ["dep:linux-keyutils"]
 apple-native = ["dep:security-framework"]
 windows-native = ["dep:windows-sys", "dep:byteorder"]
 
-sync-secret-service = ["dep:dbus-secret-service"]
-async-secret-service = ["dep:secret-service", "dep:zbus"]
+keyutils = ["dep:linux-keyutils"]
+secret-service = []
+secret-service-with-keyutils = ["secret-service", "keyutils"]
```

Comment:
If you insist on support both sync and async, then just have two features "sync-secret-service-with-keyutils" and "async-secret-service-with-keyutils", and have them enable the appropriate dependencies. So you have to add a couple of "any" conditions in the cfg logic.  Big deal. If you're adding two keystores (which you are), then have a feature for each one, and be done with it. It's the client view of the features that should be simple, not the configuration view.